rustual-boy-core/src/rom.rs

```diff
@@ -22,16 +22,22 @@ impl Rom {
         let mut vec = Vec::new();
         file.read_to_end(&mut vec)?;
 
-        let size = vec.len();
+        Rom::from_bytes(&vec)
+    }
+
+    pub fn from_bytes(bytes: &[u8]) -> io::Result<Rom> {
```

Comment:
I really like leaving the option of copying or giving away ownership to the caller though. The ergonomics are not quite there unfortunately :/